File: ios-native/NeobankNative/Sources/Features/Transfers/TransferModels.swift
import Foundation

struct Transfer: Decodable, Sendable, Equatable {
    let id: String?
    let status: String?
    let senderUserId: String?
    let recipientUserId: String?
    let amount: String?
    let currency: String?
    let failureReason: String?
    let memo: String?
    let createdAt: Date?
    let completedAt: Date?

    var isCompleted: Bool { status == "completed" }
    var isFailed: Bool { status == "failed" || status == "declined" }
}

File: android-native/app/src/main/kotlin/dev/horobets/neobanknative/features/transfers/TransferModels.kt
package dev.horobets.neobanknative.features.transfers

import kotlinx.serialization.Serializable

@Serializable
data class Transfer(
    val id: String? = null,
    val status: String? = null,
    val senderUserId: String? = null,
    val recipientUserId: String? = null,
    val amount: String? = null,
    val currency: String? = null,
    val failureReason: String? = null,
    val memo: String? = null,
    val createdAt: String? = null,
    val completedAt: String? = null,
) {
    val isCompleted: Boolean get() = status == "completed"
    val isFailed: Boolean get() = status == "failed" || status == "declined"
}
